src/Servers.jl

```diff
@@ -336,22 +324,21 @@ function listen!(f, listener::Listener;
     access_log::Union{Function,Nothing}=nothing,
     verbose=false, kw...)
     conns = Set{Connection}()
-    ready_to_accept = Ref(false)
+    ready_to_accept = Base.Event()
     if verbose > 0
         tsk = @async LoggingExtras.withlevel(Logging.Debug; verbosity=verbose) do
             listenloop(f, listener, conns, tcpisvalid, max_connections, readtimeout, access_log, ready_to_accept)
         end
     else
         tsk = @async listenloop(f, listener, conns, tcpisvalid, max_connections, readtimeout, access_log, ready_to_accept)
     end
+    # This rationale is utter bullshit, but is our fake reason for this event to exist:
```

Comment:
oof. harsh.

Reply:
```suggestion
```

Reply:
Shouldn't you delete the rest of the comment and the ready_to_accept event code simultaneously too?

Reply:
Definitely welcome to refactor things; the reason I added this was because we had all these cases in tests/scripts where people did:
```julia
@async run_localhost_server()
sleep(1)
resp = HTTP.get(localhost)
```
the `sleep` was needed because the `HTTP.get` call would return an error if the server hadn't fully started yet.

Now that I think about it, we're probably already in a better place here because I already refactored with the `Listener` piece which ensures `HTTP.listen!` doesn't return until we've already got a listening tcp server socket. So yeah, we can probably just remove this stuff.

Reply:
yeah, as soon as the socket is open, it is ready to listen, so this isn't needed